x-pack/filebeat/input/azureblobstorage/scheduler.go

```diff
@@ -192,35 +192,20 @@ func (s *scheduler) fetchBlobPager(batchSize int32) *azruntime.Pager[azblob.List
 func (s *scheduler) moveToLastSeenJob(jobs []*job) []*job {
 	var latestJobs []*job
 	jobsToReturn := make([]*job, 0)
-	counter := 0
-	flag := false
-	ignore := false
 
 	for _, job := range jobs {
 		switch {
 		case job.timestamp().After(s.state.checkpoint().LatestEntryTime):
 			latestJobs = append(latestJobs, job)
-		case job.name() == s.state.checkpoint().BlobName:
-			flag = true
 		case job.name() > s.state.checkpoint().BlobName:
-			flag = true
-			counter--
-		case job.name() <= s.state.checkpoint().BlobName && (!ignore):
-			ignore = true
+			jobsToReturn = append(jobsToReturn, job)
 		}
-		counter++
-	}
-
-	if flag && (counter < len(jobs)-1) {
-		jobsToReturn = jobs[counter+1:]
-	} else if !flag && !ignore {
-		jobsToReturn = jobs
 	}
 
 	// in a senario where there are some jobs which have a greater timestamp
 	// but lesser alphanumeric order and some jobs have greater alphanumeric order
 	// than the current checkpoint blob name, then we append the latest jobs
-	if len(jobsToReturn) != len(jobs) && len(latestJobs) > 0 {
+	if len(latestJobs) > 0 {
 		jobsToReturn = append(latestJobs, jobsToReturn...)
```

Comment:
Is the ordering here important? This is a non-conventional use of append, and the comment says that we are _appending_ the latest jobs.

Reply:
Actually the comment should be modified to appending lexicographically ordered jobs to latest jobs. The latest jobs should always be processed 1st followed by the rest. So the ordering is important but the comment is wrongly written.

Reply:
Then maybe we can get away with a single slice and a final sort?

Reply:
I'm a bit hesitant in clumping them in a single slice cause it can create debugging issues as it will obfuscate the ordering logic, having the two separate slices will help with clearer metrics and debugging.

Reply:
What I'm suggesting is something like this.
```
func (s *scheduler) moveToLastSeenJob(jobs []*job) []*job {
	latest := s.state.checkpoint().LatestEntryTime
	jobs = slices.DeleteFunc(jobs, func(j *job) bool {
		return !(j.timestamp().After(latest) || j.name() > s.state.checkpoint().BlobName)
	})
	sort.SliceStable(jobs, func(i, _ int) bool {
		return jobs[i].timestamp().After(latest)
	})
	return jobs
}
```
Basis for the segmentation is here: https://go.dev/play/p/dLeERNPD_CM (assumes order other than pivot-based is not relevant, but this can be recovered by reversing the first elements lower than the pivot).

Reply:
Ok gotcha, that makes sense. I initially assumed you were suggesting a standard sort based on the timestamp. But the sort being only pivot based makes sense here. Thanks for the clarification.